### PR TITLE
[SYCL][XPTI] Enable XPTI and XPTI Framework E2E tests

### PR DESCRIPTION
**Pull request by alexbatashev**
Add a new group of tests to ensure correct integration of XPTI tracing in SYCL runtime.

### CODE REVIEW

**Comment by alexbatashev** [comment]
Depends on https://github.com/intel/llvm/pull/4545 and https://github.com/intel/llvm/pull/4494

**Review by vladimirlaz — APPROVED** [review]

File: SYCL/lit.cfg.py [hunk 1]
```diff
@@ -313,6 +313,19 @@
 if find_executable('sycl-ls'):
     config.available_features.add('sycl-ls')
 
+# TODO properly set XPTIFW include and runtime dirs
```
Comment (vladimirlaz) [inline SYCL/lit.cfg.py:333]: The change looks like a hack.
We can detect the presence of the library in Cmaake and pass this value through lit.site.cfg.py. Also, you can get proper include and other directories from CMake configuration to resolve TODO above.

File: SYCL/XPTI/basic_event_collection.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
+// REQUIRES: xptifw
+// RUN: %clangxx -fsycl %s -DXPTI_COLLECTOR -shared -fPIC -std=c++17 -o %t_collector.dll -lxptifw
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env XPTI_TRACE_ENABLED=1 env XPTI_FRAMEWORK_DISPATCHER=%xptifw_lib env XPTI_SUBSCRIBERS=%t_collector.dll %CPU_RUN_PLACEHOLDER %t %CPU_CHECK_PLACEHOLDER
```
Comment (vladimirlaz) [inline SYCL/XPTI/basic_event_collection.cpp]: The test is running on CPU only. Is it a requirement? If so, `cpu` should be added to REQUIRES. If not we can remove CPU_RUN_PLACEHOLDERs.

As for my understanding, it should work for any BE so it is better to removed dependencies on specific device type.

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/XPTI/basic_event_collection.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
+// REQUIRES: xptifw, opencl
+// RUN: %clangxx %s -DXPTI_COLLECTOR -L%xptifw_lib -I%xptifw_includes -shared -fPIC -std=c++17 -o %t_collector.dll -lxptifw
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env XPTI_TRACE_ENABLED=1 env XPTI_FRAMEWORK_DISPATCHER=%xptifw_dispatcher env XPTI_SUBSCRIBERS=%t_collector.dll env SYCL_DEVICE_FILTER=opencl %t.out | FileCheck %s 2>&1
```
Comment (romanovvlad) [inline SYCL/XPTI/basic_event_collection.cpp]:
```suggestion
// RUN: env XPTI_TRACE_ENABLE=1 env XPTI_FRAMEWORK_DISPATCHER=%xptifw_dispatcher env XPTI_SUBSCRIBERS=%t_collector.dll env SYCL_DEVICE_FILTER=opencl %t.out | FileCheck %s 2>&1
```

XPTI_TRACE_ENABLE=1 ?

File: SYCL/lit.cfg.py [hunk 4]
```diff
@@ -313,6 +313,20 @@
...
+if platform.system() == "Linux":
+    xptifw_dispatcher = os.path.join(xptifw_lib, 'libxptifw.so')
+elif platform.system() == "Windows":
+    xptifw_dispatcher = os.path.join(config.dpcpp_root_dir, 'bin', 'xptifw.dll')
+xptifw_includes = os.path.join(config.dpcpp_root_dir, 'include')
+if os.path.exists(xptifw_lib) and os.path.exists(os.path.join(xptifw_includes, 'xpti', 'xpti_trace_framework.h')):
+    config.available_features.add('xptifw')
+    config.substitutions.append(('%xptifw_lib', xptifw_lib))
+    config.substitutions.append(('%xptifw_dispatcher', xptifw_dispatcher))
+    config.substitutions.append(('%xptifw_includes', xptifw_includes))
```
Comment (vladimirlaz) [inline SYCL/lit.cfg.py]: Do we really need separate substitutions? Are they used separately?
if not we can merge them

File: SYCL/XPTI/basic_event_collection.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
+// REQUIRES: xptifw, opencl
+// RUN: %clangxx %s -DXPTI_COLLECTOR -L%xptifw_lib -I%xptifw_includes -shared -fPIC -std=c++17 -o %t_collector.dll -lxptifw
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (vladimirlaz) [inline SYCL/XPTI/basic_event_collection.cpp:3]:
```suggestion
// RUN: %clangxx -fsycl %s -o %t.out
```
If options are not needed we can skip them

File: SYCL/XPTI/basic_event_collection.cpp [hunk 6]
```diff
@@ -0,0 +1,125 @@
+// REQUIRES: xptifw, opencl
+// RUN: %clangxx %s -DXPTI_COLLECTOR -L%xptifw_lib -I%xptifw_includes -shared -fPIC -std=c++17 -o %t_collector.dll -lxptifw
```
Comment (vladimirlaz) [inline SYCL/XPTI/basic_event_collection.cpp]: the command line will not work is clang-cl is used (-std=c++17 and -fPIC will not work).

There ar substitutions for such a case: https://github.com/intel/llvm-test-suite/blob/intel/SYCL/lit.cfg.py#L139

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

**Comment by bader** [comment]
@alexbatashev, please, resolve merge conflicts and apply suggestions from previous reviews.

**Review by vladimirlaz — DISMISSED** [review]

**Review by vladimirlaz — DISMISSED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**